### PR TITLE
Replace comments with whitespace by default

### LINKED ISSUE #24: Replace comments with whitespace
Just a thought I got from TypeScript actually, but what about replacing the comments with whitespace so that tooling will report the correct error positions?

### LINKED ISSUE #26: Breaks Regular Expressions
Examples of valid RegExp that gets broken:

```js
// thinks it is a text opening:
/'/
/"/
/\`/
```

```js
// thinks it is a single-line comment opening:
/\//
```

```js
// thinks it is a multi-line comment opening:
/\/*/
```

### PR DESCRIPTION
**Pull request by blakeembrey**
Closes #24. Implements the output using slices of text instead of character by character.

### CODE REVIEW

File: strip-json-comments.js [hunk 1]
```diff
@@ -11,12 +11,23 @@
...
+	function sliceWithoutWhitespace (str, start, end) {
+		return '';
+	}
+
+	function sliceWithWhitespace (str, start, end) {
+		return str.slice(start, end).replace(/\S/g, ' ');
+	}
+
+	function stripJsonComments(str, opts) {
+		var options = opts || {};
```
Comment (kevva) [inline strip-json-comments.js]: Maybe just `opts = opts || {}`?

**Comment by kevva** [comment]
Looks really good!

**Comment by blakeembrey** [comment]
@kevva Updated with your comment.

**Comment by sindresorhus** [comment]
@blakeembrey Can you document the `whitespace` option?

**Comment by blakeembrey** [comment]
@sindresorhus Done. Let me know if it needs to be more explicit, maybe documenting what happens on `false`.

**Comment by sindresorhus** [comment]
> maybe documenting what happens on false.

Yeah

**Comment by blakeembrey** [comment]
Fixed :+1:

**Comment by blakeembrey** [comment]
@sindresorhus Ping.

**Comment by sindresorhus** [comment]
Thank you! Looks good :)

Sorry about the delay. I'm usually faster, but had especially many OSS things to attend to lately.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/strip-json-comments/releases/tag/v2.0.0

**Pull request merged**